src/Microsoft.Data.SqlClient/netcore/src/Microsoft.Data.SqlClient.csproj

```diff
@@ -393,6 +394,8 @@
     <Compile Include="Microsoft\Data\SqlClient\SNI\SslOverTdsStream.NetCoreApp.cs" />
     <Compile Include="Microsoft\Data\SqlClient\SqlConnectionFactory.AssemblyLoadContext.cs" />    
     <Compile Include="Microsoft\Data\SqlClient\SqlDependencyUtils.AssemblyLoadContext.cs" />
+    <Compile Condition="$(TargetFramework.StartsWith('netcoreapp2.'))" Include="Microsoft\Data\SqlClient\SNI\SNIStreams.Task.cs" />
+    <Compile Condition="!$(TargetFramework.StartsWith('netcoreapp2.'))" Include="Microsoft\Data\SqlClient\SNI\SNIStreams.ValueTask.cs" />
```

Comment:
This should also be handled for **.NET Standard 2.0 (netstandard2.0)**, since .NET Core 2.1 can be targeted via .NET Standard 2.0 DLL too.

Also for context, we will be removing .NET Core 2.1 DLL from NuGet at some point in future as it's going out of mainstream support soon, but .NET Standard 2.0 DLL will continue to work for .NET Core 2.1.

Reply:
The netstandard inclusion is done with the task version on line 347 above.

Reply:
Ok, so then we need to address it for .NET Standard 2.1. Since we have 2 different DLLs:
- **netstandard2.0** (also applies indirectly to .NET Core 2.1+)
- **netstandard2.1** (also applies indirectly to .NET Core 3.1+)
 
And [ValueTask](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.valuetask-1?view=net-5.0) is available in netstandard2.1, so we should provide it there IMO.

Reply:
It depends on the exact implementation. If you tried to use the ValueTask version on netcore2.1, which is possible to do because it's netstandard 2.0 compliant, it would end up in the infinite loop that I fixed earlier on because while ValueTask is supported the _implementation_ on netcore2.1 forwards valuetask to task. It's not safe to use ValueTask unless you know the runtime supports it directly. 

The current setup is safe. Anything else you'll need to check independantly.

Reply:
We can leave it too, there's no requirement or blocking concern - it just creates separate paths for .NET Standard 2.1 and .NET Core 3.1, diverging their code further. But I agree unknown cases can occur with other targets.